### ISSUE TITLE
Questions about spine curve

### ISSUE THREAD

**Issue by Lindarun** [neuro-ml/straighten#1]
Hello, When I have finished reading your paper "Interpretable Vertebral Fracture Quantification via Anchor-Free Landmarks Localization," and I have some questions to inquire about:
1. What is the data format for the 3D curve of the spine as proposed by the paper? Does it refer to a set of coordinates for points in three-dimensional space?
2. If it refers to a set of coordinates for points in three-dimensional space, can I use the centroid coordinates of the vertebraes instead?

I will appreciate your help and insightful response. :)

**Comment by maxme1** [comment]
Hi. I'm glad you read our paper!
The curve has 2 representations, depending on what you mean:
1. the original curve is a list of 3D points, denoting the knots of the curve. Basically, this is  a numpy array of shape `(N, 3)` (say `knots`). Note that here the list is always ordered along the last (z) coordinate: `knots[i, 2] < knots[i+1, 2]`. This is because the spine is a, more or less, vertical curve.
2. at train time we encode it a bit differently: we interpolate the curve, so that `curve[i, 2] == i`. This means that each knot lies on the corresponding horizontal plane. Next, we simply drop the last coordinate, as it is now redundant.

We need the second step, because we compute a loss function, and the `knots` are too "arbitrary" for this, there are many sets of knots that encode the same curve. While in the second case, the representation is unique (and "local").

>  can I use the centroid coordinates of the vertebraes instead

Yes, this is exactly how we obtained the `knots` for the VerSe dataset in our paper.

By the way, if you need a simple interface for reading the data from our paper, you should check out [AMID](https://github.com/neuro-ml/amid) - our datasets library.

Hope it clarifies a bit. If not, please, feel free to ask more questions :slightly_smiling_face:

**Comment by Lindarun** [comment]
> Hi. I'm glad you read our paper! The curve has 2 representations, depending on what you mean:
> 
> 1. the original curve is a list of 3D points, denoting the knots of the curve. Basically, this is  a numpy array of shape `(N, 3)` (say `knots`). Note that here the list is always ordered along the last (z) coordinate: `knots[i, 2] < knots[i+1, 2]`. This is because the spine is a, more or less, vertical curve.
> 2. at train time we encode it a bit differently: we interpolate the curve, so that `curve[i, 2] == i`. This means that each knot lies on the corresponding horizontal plane. Next, we simply drop the last coordinate, as it is now redundant.
> 
> We need the second step, because we compute a loss function, and the `knots` are too "arbitrary" for this, there are many sets of knots that encode the same curve. While in the second case, the representation is unique (and "local").
> 
> > can I use the centroid coordinates of the vertebraes instead
> 
> Yes, this is exactly how we obtained the `knots` for the VerSe dataset in our paper.
> 
> By the way, if you need a simple interface for reading the data from our paper, you should check out [AMID](https://github.com/neuro-ml/amid) - our datasets library.
> 
> Hope it clarifies a bit. If not, please, feel free to ask more questions 🙂

Thank you for your prompt and detailed response! I appreciate your clarification regarding the representations of the curve.  And I will make sure to explore AMID the datasets library you mentioned, as it could be very useful for my work.

If any further inquiries arise, I won't hesitate to reach out. 🙂

Best regards
